toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon

```diff
@@ -369,6 +369,23 @@ fn F() {
   }
 }
 
+// --- fail_implicitly_poison_core.carbon
```

Comment:
Why is this in `testdata/function/declaration/...`? It doesn't really seem to be testing function declarations.

Reply:
You're absolutely right, there a lot of tests here that are definitely not about function declaration and/or not specific to function declaration.
I thought of putting all name poisoning tests together, but I agree this is not a good place.
I think at least some of the tests don't really belong to any specific existing testdata directory.
Should I create a dedicated one? Perhaps "lookup"?
Will probably do this in a separate PR though if that's ok.

Reply:
#4909 has `testdata/packages/raw_core.carbon`. You're testing name poisoning of the package; how about either that file or `testdata/pavckages/core_name_poisoning.carbon`?

Reply:
I've moved it to `toolchain/check/testdata/packages/no_prelude/core_name_poisoning.carbon`.

Reply:
Thanks! I see you requested another review from zygoloid, but I think this is small enough to go ahead with a merge.